src/codegen/sdk/core/file.py

```diff
@@ -949,62 +944,56 @@ def update_filepath(self, new_filepath: str) -> None:
             imp.set_import_module(new_module_name)
 
     @writer
-    def add_symbol_import(
-        self,
-        symbol: Symbol,
-        alias: str | None = None,
-        import_type: ImportType = ImportType.UNKNOWN,
-        is_type_import: bool = False,
-    ) -> Import | None:
-        """Adds an import to a file for a given symbol.
-
-        This method adds an import statement to the file for a specified symbol. If an import for the
-        symbol already exists, it returns the existing import instead of creating a new one.
-
-        Args:
-            symbol (Symbol): The symbol to import.
-            alias (str | None): Optional alias for the imported symbol. Defaults to None.
-            import_type (ImportType): The type of import to use. Defaults to ImportType.UNKNOWN.
-            is_type_import (bool): Whether this is a type-only import. Defaults to False.
-
-        Returns:
-            Import | None: The existing import for the symbol or None if it was added.
-        """
-        imports = self.imports
-        match = next((x for x in imports if x.imported_symbol == symbol), None)
-        if match:
-            return match
-
-        import_string = symbol.get_import_string(alias, import_type=import_type, is_type_import=is_type_import)
-        self.add_import_from_import_string(import_string)
-
-    @writer(commit=False)
-    def add_import_from_import_string(self, import_string: str) -> None:
-        """Adds import to the file from a string representation of an import statement.
+    def add_import(self, imp: Symbol | str, *, alias: str | None = None, import_type: ImportType = ImportType.UNKNOWN, is_type_import: bool = False) -> Import | None:
```

Comment:
Can we add typing overloads to make sure the user doesn't pass the other parameters if imp is of type str